### DISCUSSION TITLE
Get the right KNN distances matrix

### DISCUSSION THREAD

**Discussion by QuKunLab** [satijalab/seurat#5339] [Q&A]
Hi, seurat team,
I have got a question when I'd like to export the KNN distances matrix, I noticed the seurat object only stores these two information: neighbors matrix(which contains 0, 1) and connectivity matrix.So how can i get the  KNN distances matrix from seurat?

**Comment by andrewwbutler** [comment]
Hi,

If you provide `return.neighbor = TRUE` to `FindNeighbors`, this will save the KNN distance matrix which can then be accessed with `Distances(ob[["X"]])`, where `X` is the name of the neighbor object (defaults to `Assay.nn`)
